influxdb/src/client/mod.rs

```diff
@@ -23,18 +23,30 @@ use std::fmt::{self, Debug, Formatter};
 use std::sync::Arc;
 #[cfg(feature = "surf")]
 use surf::{Client as HttpClient, RequestBuilder, Response as HttpResponse};
+use std::marker::PhantomData;
 
 use crate::query::QueryType;
 use crate::Error;
 use crate::Query;
 
+/// Marker type for InfluxDB Version 1
+#[derive(Clone)]
+pub struct InfluxVersion1;
+/// Marker type for InfluxDB Version 2
+#[derive(Clone)]
+pub struct InfluxVersion2;
+/// Marker type for InfluxDB Version 3
+#[derive(Clone)]
+pub struct InfluxVersion3;
+
 #[derive(Clone)]
 /// Internal Representation of a Client
-pub struct Client {
+pub struct Client<V, H = reqwest::Client> {
     pub(crate) url: Arc<String>,
     pub(crate) parameters: Arc<HashMap<&'static str, String>>,
     pub(crate) token: Option<String>,
-    pub(crate) client: HttpClient,
+    pub(crate) client: H,
```

Comment:
I don't think this change makes much sense, given that we already decided to sunset surf support in https://github.com/influxdb-rs/influxdb-rust/issues/147

Reply:
Yeah, I was thinking of doing #147 first.